File: boilerplate/nestjs/order-service/src/infrastructure/persistence/user.entity.ts
import { Entity, PrimaryColumn, Column } from "typeorm";

import { Role } from "../../domain/models/role";

@Entity("users")
export class User {
  @PrimaryColumn()
  id: string;

  @Column({ unique: true })
  email: string;

  @Column({ name: "password_hash" })
  passwordHash: string;

  @Column("simple-array")
  roles: Role[];

  @Column({ default: true })
  enabled: boolean;

  @Column({ name: "created_at" })
  createdAt: Date;

  @Column({ name: "last_login_at", nullable: true })
  lastLoginAt: Date;
}

File: boilerplate/java/order-service/src/main/java/com/example/orderservice/infrastructure/persistence/UserEntity.java
package com.example.orderservice.infrastructure.persistence;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "users")
@Getter
@Setter
@NoArgsConstructor
public class UserEntity {

    @Id
    private UUID id;

    private String email;

    private String passwordHash;

    private String roles;

    private boolean enabled;

    private OffsetDateTime createdAt;

    private OffsetDateTime lastLoginAt;

    public UserEntity(UUID id, String email, String passwordHash, String roles,
                      boolean enabled, OffsetDateTime createdAt, OffsetDateTime lastLoginAt) {
        this.id = id;
        this.email = email;
        this.passwordHash = passwordHash;
        this.roles = roles;
        this.enabled = enabled;
        this.createdAt = createdAt;
        this.lastLoginAt = lastLoginAt;
    }
}
